lang/c/src/datum.c

```diff
@@ -153,7 +155,8 @@ static avro_datum_t avro_bytes_private(char *bytes, int64_t size,
 	return &datum->obj;
 }
 
-avro_datum_t avro_bytes(const char *bytes, int64_t size)
+avro_datum_t avro_bytes(avro_schema_t schema,
+                        const char *bytes, int64_t size)
```

Comment:
this breaks backwards compatibility with the legacy api: stragglers that haven't updated their codebase to the current api probably won't be eager to do so for the sake of logical schema support